domains/noto/internal/noto/states.go

```diff
@@ -199,16 +211,55 @@ type preparedLockedInputs struct {
 }
 
 type preparedOutputs struct {
-	coins  []*types.NotoCoin
-	states []*prototk.NewState
+	distributions []identityList
+	coins         []*types.NotoCoin
+	states        []*prototk.NewState
 }
 
 type preparedLockedOutputs struct {
-	coins  []*types.NotoLockedCoin
-	states []*prototk.NewState
+	distributions []identityList
+	coins         []*types.NotoLockedCoin
+	states        []*prototk.NewState
+}
+
+type identityPair struct {
```

Comment:
This is a struct we tend to need often... I've been using it in lots of apps and happened to pick the name "resolved identity". No need to change anything here, but just an observation that it seems to be a common pattern of Paladin development.